octokit/gists_test.go

```diff
@@ -151,34 +145,31 @@ func TestGistsService_Update(t *testing.T) {
 	setup()
 	defer tearDown()
 
-	params := Gist{}
-	params.Description = "the description for this gist"
-	params.Public = true
-
-	var gistFile1 = GistFile{}
-	gistFile1.Content = "updated file contents"
-	var gistFile2 = GistFile{}
-	gistFile2.FileName = "new_name.txt"
-	gistFile2.Content = "modified contents"
-	var gistFile3 = GistFile{}
-	gistFile3.Content = "a new file"
-
-	params.Files = map[string]*GistFile{"file1.txt": &gistFile1, "old_name.txt": &gistFile2,
-		"new_file.txt": &gistFile3, "delete_this_file.txt": nil}
-
-	mux.HandleFunc("/gists", func(w http.ResponseWriter, r *http.Request) {
-		var gistParams Gist
-		json.NewDecoder(r.Body).Decode(&gistParams)
-		assert.Equal(t, params.Description, gistParams.Description)
-		assert.Equal(t, params.Public, gistParams.Public)
-		assert.Equal(t, params.Files, gistParams.Files)
-
-		assert.Len(t, params.Files, 4)
-
-		testMethod(t, r, "PATCH")
-		respondWithJSON(w, loadFixture("gist.json"))
-	})
-
+	wantReqBody := fmt.Sprintf(`{"%s":"%s",`+
```

Comment:
@feiranchen: Would [`json.Marshal`][marshal] be a clearer approach here? 

[marshal]: http://golang.org/pkg/encoding/json/#Marshal